File: JavaScript/0137_single_number_II/solution.js
function single_number(nums) {
    let num_count = {};
    for (let num of nums) {
        if (num in num_count) {
            num_count[num] ++;
        } else {
            num_count[num] = 1;
        }
    }
    for (let [key,value] of Object.entries(num_count)) {
        if (value == 1) {
            return key;
        }
    }
    return null;
}

const nums = [0,1,0,1,0,1,99];
console.log(single_number(nums));

File: Python/0137_single_number_II/solution.py
def single_number(nums):
    num_count = dict()
    for num in nums:
        if num not in num_count:
            num_count[num] = 1
        else:
            num_count[num] += 1
    for key,value in num_count.items():
        if value == 1:
            return key
    return None

nums = [0,1,0,1,0,1,99]
print(single_number(nums))